Build issues

LSchwiebert opened the pull request:
This is a major reworking of the build process and some related changes to the core non-functional parts of the GOMC repository. The changes include:
1. A major edit of the Readme file, including adding the DOI to each of the two references. And clarification of some of the text.
2. Adding missing copyright notices or correcting ones that were outdated.
3. Removal of a file that is no longer used (Endian.h) and removed the corresponding google test files.
4. Changes to the build process to add flags to make it easier to customize the build and avoid conflicts among flags. For instance, you now use -t to run clang-tidy and use -a to build with address sanitizer. Also adds support for the Intel LLVM and Clang compilers.
5. Major rewrite of the CMake process to include link-time optimization for builds and setting other optimization flags. Should result in better performance. These optimizations can be skipped by using the -n flag with metamake.sh. (Removed the link-time optimization flags because CMake doesn't seem to play well with the GPU LTO flags.)
6. Cleaned up the headers to use consistent naming and end with an appropriate comment. Some retained information from other header files.

This is a necessary step before our next release, so please test the new build process carefully with available compilers.

LSchwiebert committed:
Improve build process and minor build-related code updates

LSchwiebert committed:
Correct EOL

LSchwiebert committed:
Fix a minor memory leak

LSchwiebert committed:
Make sure the correct abs() function is used with floats

LSchwiebert committed:
Update for recent code changes

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Add support for Clang on Linux

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Restore to development version

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Fixes the gcc compiler issues but not linker

LSchwiebert committed:
Final changes to resolve GPU linking issues

LSchwiebert committed:
Update GOMC Website in Readme file

LSchwiebert committed:
Update web link for CMake

LSchwiebert committed:
Remove unused use_tidy variable. Just set CMake flag

LSchwiebert committed:
Clean up README.md per Codacy style

LSchwiebert committed:
README.md clean up per Codacy Style guidelines

LSchwiebert committed:
README.md clean up per Codacy Style guidelines

GregorySchwing reviewed (commented):
Looks good.  Still need to clone and try, but why are you removing openMP flags?  Is this no longer necessary, or did you move it?

# find OpenMP and set it up
find_package(OpenMP)
if (OPENMP_FOUND)
	set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
	set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
endif()

LSchwiebert commented:
Thanks for taking a look. It's still there. I just needed to move it earlier in the CMakeLists.txt file so it was defined before I set the compiler flags. It looks like it was removed, but it's there starting on line 56.

I did turn off building with OpenMP if Address Sanitizer is being used, because they aren't compatible, but no one should be building with address sanitizer except to debug code or out of curiosity.

LSchwiebert committed:
Change OpenMP pragma for compatibility with old Intel compilers

LSchwiebert committed:
Change OpenMP pragma for compatibility with old Intel compilers

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'development' into build-issues

LSchwiebert committed:
Update CMakeLists.txt for handling clang-tidy

LSchwiebert committed:
Merge branch 'build-issues' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Remove Endian Tests since Endian.h is no longer used in GOMC

LSchwiebert committed:
Standardize guards on header files. Add some missing ones.

LSchwiebert committed:
Print error message if CUDA does not support this compiler

LSchwiebert committed:
Build test executables only for specified ensembles

LSchwiebert committed:
Warning instead Status message for compiler incompatible with CUDA

LSchwiebert committed:
Standardize guards on header files

LSchwiebert committed:
Fix syntax error in CMakeLists.txt for warning message

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Remove link time optimization from the build process

LSchwiebert commented:
I've completed the patch, so it's ready for testing. I don't have access to a newer Intel compiler, so would like to know what happens when building the GPU code with the new ipcx compiler.

GregorySchwing committed:
first commit

GregorySchwing committed:
Everything except FFParticle, ect

GregorySchwing committed:
ok should be testable

GregorySchwing committed:
read wolf alpha

GregorySchwing committed:
init wolf

GregorySchwing committed:
Add calibration infra

GregorySchwing committed:
straigtforward implementation of wolf calc

GregorySchwing committed:
actually call wco

GregorySchwing committed:
wrapping up cal

GregorySchwing committed:
Calc on first step

GregorySchwing committed:
update ewald ptr after swapping

GregorySchwing committed:
was missing qqFactor in BoxSelf

GregorySchwing committed:
dont use sample in WolfCal

GregorySchwing committed:
Update readme with docker build instructions.  will add a section to pull from dockerhub later

GregorySchwing committed:
all 8 methods.

GregorySchwing committed:
methodized simple corr

GregorySchwing committed:
for some reason no difference between dsp and dsf?

GregorySchwing committed:
done

GregorySchwing committed:
indent

GregorySchwing committed:
GPU

GregorySchwing committed:
energy closer.  need to do last function

GregorySchwing committed:
all functions are done except for the actual code

GregorySchwing committed:
energy done

GregorySchwing committed:
energy done

GregorySchwing committed:
remove 35 arch

GregorySchwing committed:
Revert "remove 35 arch"

This reverts commit 5b2a065c0c6f9b72fde674d20cba14978bcc46cd.

GregorySchwing committed:
actually use simple self

LSchwiebert committed:
Remove Intel LLVM Compiler detection until CUDA supports it

LSchwiebert committed:
Clean up some #define statements and comments on the #endif for header files

LSchwiebert commented:
I removed the use of the new Intel LLVM compiler until CUDA supports it. This patch should be ready for testing and merging.

Most of the changes are cosmetic -- general code cleanup. For example, I updated the README file, added missing copyright notices to 3rd party software that we are using, standardized the #define for including header files, including fixing some that didn't match the filename, improve the build process, especially for the code, and removed unused files.

Once this is approved, we can proceed with the next GOMC release.

LSchwiebert committed:
Resolve Codacy-detected error for builds with test

LSchwiebert committed:
Handle ensemble names when building the test configs

LSchwiebert committed:
Build CUDA code for GPU executables

LSchwiebert committed:
Don't include GPU calls in CPU builds

LSchwiebert committed:
Don't call RotationMatrix::UniformRandom with prng() calls as function parameters so that gcc and intel give consistent results

LSchwiebert committed:
Remove old UniformRandom function calls

LSchwiebert committed:
Fix typo in comment

LSchwiebert committed:
Add entries to print newer OpenMP versions

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Patch to handle older OpenMP versions

LSchwiebert committed:
Updates to build on Windows with Visual Studio

LSchwiebert committed:
Clarify Windows build

LSchwiebert committed:
Updates to resolve MSVC warnings and improve MSVC build flags

LSchwiebert committed:
Pass more forcefield parameters to the GPU for consistent values

LSchwiebert committed:
Reformat source files using clang format

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Update for latest development branch patches

LSchwiebert committed:
Set address sanitizer MSVC flag only for MSVC builds

LSchwiebert committed:
Fix printf format to match variable types

LSchwiebert committed:
Update with the latest Development branch patches

LSchwiebert committed:
Print clearer logfile information on CPU cores and OpenMP threads

LSchwiebert committed:
Inline some commonly used MinImage functions for performance

LSchwiebert committed:
Improve Box Dimensions output formatting for large box sizes

LSchwiebert committed:
Resolve warnings on virtual functions only partially overridden

LSchwiebert committed:
Call correct template function for Brownian Motion MP rotation with nonorthogonal box

LSchwiebert committed:
Reset fp format to previous settings after printing energies in debug mode

LSchwiebert committed:
Choose better CUDA debug flags

LSchwiebert committed:
Avoid subnormal values in DCRotations and Random PickWeighted

LSchwiebert committed:
Make architecture-specific optimizations optional instead of default

LSchwiebert committed:
Fix extraneous text that broke build

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Fix list of options when parsing commandline flags

LSchwiebert committed:
Resolve compiler warnings when not compiling with MPI

LSchwiebert committed:
Change to support CUDA 13

LSchwiebert committed:
Reformat per clang format

LSchwiebert committed:
Clear CUDA flag if no GPU builds selected

LSchwiebert committed:
Require C++17

LSchwiebert committed:
Set CMake CUDA architectures based on CUDA version

LSchwiebert committed:
Set GPU architectures based on CUDA version

LSchwiebert committed:
Enable unsupported intel compilers with CUDA

LSchwiebert committed:
Streamline setting of CUDA compiler flags

LSchwiebert committed:
Better test for CUDA compiler version for GPU builds

LSchwiebert committed:
Fix Codacy code warnings

LSchwiebert committed:
Update to cereal 1.3.2 and include only needed cereal files

LSchwiebert committed:
Rename cereal folder without version and move up in lib folder

LSchwiebert committed:
Free three additional GPU variables

LSchwiebert committed:
Fix erroneous improvement to LJ weight calculations

LSchwiebert committed:
Remove unnecessary setting of ljweights in BuildOld

LSchwiebert committed:
Ensure Steps/sec gets written with the console output

LSchwiebert committed:
Include -lineinfo flag when building for CUDA profiling

LSchwiebert committed:
Patches to enable profiling builds

LSchwiebert committed:
Fix processing of GOMC-supplied profiling event list

LSchwiebert committed:
Partial fix of breaking MPI build

LSchwiebert committed:
Make intel use more precise arithmetic

LSchwiebert committed:
Update to Random123 Version 1.14 with our patch for CUDA

LSchwiebert committed:
Revise intel compiler flags to prevent overly aggressive math optimizations

LSchwiebert committed:
Remove architecture specific flags that don't support AMD processors

LSchwiebert committed:
Choice GPU based on number of SMs and minor code cleanup

LSchwiebert committed:
Minor code cleanup for enabling NVTX profiling

LSchwiebert committed:
Cleanup per new version of clang format

LSchwiebert committed:
Don't swap reciprocal vectors if reciprocal energy unchanged

LSchwiebert committed:
Remove unneeded computations when overlap has been detected

LSchwiebert committed:
Sort CMake file list in alphabetical order

LSchwiebert committed:
Fix typos in comments

LSchwiebert committed:
Improved short circuit of computation when overlap detected

LSchwiebert committed:
Revised formatting per new release of clang format

LSchwiebert committed:
Revert bad code optimization in PickWeighted and use signed int loop indices

LSchwiebert committed:
Remove header file <limits> that is no longer needed

LSchwiebert committed:
Change version number to 2.80 in anticipation of next release

LSchwiebert committed:
Update Readme with pointers to the examples repo and some clarifications

GregorySchwing committed:
compiles

GregorySchwing committed:
added checkpointfreq

GregorySchwing committed:
fix how tests are built

GregorySchwing committed:
mpi tests build

GregorySchwing committed:
fix codacy

GregorySchwing committed:
only set variable in mpi constructor

LSchwiebert committed:
Merge pull request #541 from GregorySchwing/pt_fix

MPI fix

LSchwiebert committed:
Update MPI package find per CMake deprecated function warning

LSchwiebert committed:
Remove redundant metamakeMPI shell script

LSchwiebert committed:
Reformat per clang format

GregorySchwing committed:
Merge branch 'wolf_fitz' into build-issues

GregorySchwing committed:
Revert "Merge branch 'wolf_fitz' into build-issues"

This reverts commit 07d3c5ae9d2a0f7efbdda286a4938036d8d68e8d, reversing
changes made to f9bef6336e355a5ef8e5c0c8f6716226bc4b0343.

LSchwiebert committed:
Merge branch 'development' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Merge branch 'build-issues' of https://github.com/GOMC-WSU/GOMC into build-issues

LSchwiebert committed:
Update change log and copyright notice for new release

LSchwiebert committed:
Clean up Codacy warnings

LSchwiebert commented:
Updates have been completed, so please approve so it can be merged into the development branch for the next release.

LSchwiebert committed:
Update per clang-format
